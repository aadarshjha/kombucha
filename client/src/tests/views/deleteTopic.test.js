// Test the Events component
import ReactDOM from "react-dom";
import DeleteTopic from "../../views/login/deleteTopic";
import renderer from "react-test-renderer";

window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      addListener: function () {},
      removeListener: function () {},
    };
  };

// ensures that the events.tsx can itself render.
it("renders DeleteTopic.tsx without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<DeleteTopic />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// checks if the form is rendered
it("renders the DeleteTopic form", () => {
  const div = document.createElement("div");
  ReactDOM.render(<DeleteTopic />, div);
  expect(div.querySelector(".centeredForm")).toBeTruthy();
});

it("renders three form fields", () => {
  const div = document.createElement("div");
  ReactDOM.render(<DeleteTopic />, div);
  expect(div.querySelectorAll(".ant-form-item-label").length).toEqual(1);
});

it("renders the correct form fields", () => {
  const div = document.createElement("div");
  ReactDOM.render(<DeleteTopic />, div);
  expect(
    [...div.querySelectorAll(".ant-form-item-label>label")].map(
      (element) => element.textContent
    )
  ).toEqual(["Topic Name"]);
});

it("renders the submit button", () => {
  const div = document.createElement("div");
  ReactDOM.render(<DeleteTopic />, div);
  expect(div.querySelectorAll(".ant-btn-primary")).toBeTruthy();
});

it("matches learn page snapshot", () => {
  const tree = renderer.create(<DeleteTopic />).toJSON();
  expect(tree).toMatchSnapshot();
});
