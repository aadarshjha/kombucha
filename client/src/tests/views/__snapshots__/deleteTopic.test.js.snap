// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches learn page snapshot 1`] = `
<div
  className="centeredForm"
>
  <form
    className="ant-form ant-form-horizontal"
    id="register"
    onSubmit={[Function]}
  >
    <div
      className="ant-row ant-form-item"
      style={
        Object {
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
        }
      }
    >
      <div
        className="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-8"
        style={Object {}}
      >
        <label
          className="ant-form-item-required"
          htmlFor="register_name"
          title="Topic Name"
        >
          Topic Name
        </label>
      </div>
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <div
              className="ant-select ant-select-single ant-select-allow-clear ant-select-show-arrow"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
            >
              <div
                className="ant-select-selector"
                onClick={[Function]}
                onMouseDown={[Function]}
              >
                <span
                  className="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="register_name_list_0"
                    aria-autocomplete="list"
                    aria-controls="register_name_list"
                    aria-haspopup="listbox"
                    aria-owns="register_name_list"
                    autoComplete="off"
                    className="ant-select-selection-search-input"
                    id="register_name"
                    onChange={[Function]}
                    onCompositionEnd={[Function]}
                    onCompositionStart={[Function]}
                    onKeyDown={[Function]}
                    onMouseDown={[Function]}
                    onPaste={[Function]}
                    readOnly={true}
                    role="combobox"
                    style={
                      Object {
                        "opacity": 0,
                      }
                    }
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  className="ant-select-selection-placeholder"
                >
                  Select a option and change input text above
                </span>
              </div>
              <span
                aria-hidden={true}
                className="ant-select-arrow"
                onMouseDown={[Function]}
                style={
                  Object {
                    "WebkitUserSelect": "none",
                    "userSelect": "none",
                  }
                }
                unselectable="on"
              >
                <span
                  aria-label="down"
                  className="anticon anticon-down ant-select-suffix"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
      style={
        Object {
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
        }
      }
    >
      <div
        className="ant-col ant-form-item-control ant-col-xs-24 ant-col-xs-offset-0 ant-col-sm-16 ant-col-sm-offset-8"
        style={Object {}}
      >
        <div
          className="ant-form-item-control-input"
        >
          <div
            className="ant-form-item-control-input-content"
          >
            <button
              className="ant-btn ant-btn-primary"
              onClick={[Function]}
              type="submit"
            >
              <span>
                Delete Topic
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
